src/utils.ts

```diff
@@ -77,13 +81,26 @@ export function getContentApiUrl(context?: MinimalContext): string {
     return mode === RENDER_MODE.NEXT ? CONTENT_API_MASTER : CONTENT_API_DRAFT;
 }
 
-export function getJsessionHeaders(context?: MinimalContext): Object {
-    const headers = {};
+export function addJsessionHeaders(headers: Object = {}, context?: MinimalContext): void {
     const jsessionid = context?.req?.headers[JSESSIONID_HEADER];
     if (jsessionid) {
         headers['Cookie'] = `${JSESSIONID_HEADER}=${jsessionid}`;
     }
-    return headers;
+}
+
+export function addLocaleHeaders(headers: Object = {}, context?: Context): void {
+    const locale = context?.locale;
+    if (locale) {
+        headers[LOCALE_HEADER] = locale;
+    }
+    const locales = context?.locales;
+    if (locales) {
+        headers[LOCALES_HEADER] = JSON.stringify(locales);
+    }
+    const defaultLocale = context?.defaultLocale;
+    if (defaultLocale) {
+        headers[DEFAULT_LOCALE_HEADER] = defaultLocale;
+    }
```

Comment:
Can there be all three headers at the same time? If not, then we should add `return`s.